tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dsl/DSLResultTest.java

```diff
@@ -0,0 +1,104 @@
+package org.dataflowanalysis.analysis.tests.dsl;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.dataflowanalysis.analysis.DataFlowConfidentialityAnalysis;
+import org.dataflowanalysis.analysis.core.AbstractVertex;
+import org.dataflowanalysis.analysis.core.FlowGraphCollection;
+import org.dataflowanalysis.analysis.dsl.AnalysisConstraint;
+import org.dataflowanalysis.analysis.dsl.AnalysisQuery;
+import org.dataflowanalysis.analysis.dsl.query.QueryDSL;
+import org.dataflowanalysis.analysis.dsl.result.DSLResult;
+import org.dataflowanalysis.analysis.dsl.variable.ConstraintVariable;
+import org.dataflowanalysis.analysis.dsl.constraint.ConstraintDSL;
+import org.dataflowanalysis.analysis.dsl.selectors.Intersection;
+import org.dataflowanalysis.analysis.dsl.selectors.CharacteristicsSelectorData;
+import org.dataflowanalysis.analysis.dsl.selectors.DataCharacteristicsSelector;
+import org.dataflowanalysis.analysis.dsl.selectors.VertexCharacteristicsSelector;
+import org.dataflowanalysis.analysis.dsl.variable.ConstraintVariableReference;
+import org.dataflowanalysis.analysis.pcm.core.user.UserPCMVertex;
+import org.dataflowanalysis.analysis.pcm.dsl.PCMVertexType;
+import org.dataflowanalysis.analysis.tests.BaseTest;
+import org.dataflowanalysis.analysis.tests.constraint.data.ConstraintData;
+import org.dataflowanalysis.analysis.tests.constraint.data.ConstraintViolations;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+
+public class DSLResultTest extends BaseTest {
+    private static final Logger logger = Logger.getLogger(DSLResultTest.class);
+
+    @Test
+    public void testDSL() {
+        var constraint = new ConstraintDSL()
+                .ofData()
+                .withLabel("DataSensitivity", List.of("Personal"))
+                .ofNode()
```

Comment:
What is the semantics `ofNode` if I add additional selectors here? In my testing, I added PCMVertxType User, because I thought, the personal data originates from a usage scenario. However, no violations were found and I had to change it to SEFF. Am I wrong here?

Reply:
The of node selector is a constraint on the attributes of the node the data _is at_. In writing this, however, I realize that constraints here could be expressed after the `neverFlows()` as well.

In the next meeting, we should talk about the syntax here

Reply:
Result of the discussion: Call it `fromNode` (or vertex?), make it transitive, does not affect `ofNode` in queries

Reply:
Result from the discussion: Vertex and label